### PR TITLE
feat(docs): add docs for Angular

### PR DESCRIPTION
**Pull request by dhmlau**
Signed-off-by: Diana Lau <dhmlau@ca.ibm.com>

Add instructions in how to build an Angular application from OpenAPIs coming from LoopBack application. 
Related to https://github.com/strongloop/loopback-next/issues/4849

Local preview:
Screen Shot 2020-10-05 at 3 03 47 PM

It's based on my blog:
https://medium.com/@MobileDiana/building-a-frontend-application-with-loopback4-and-angular-e15c6fb48971

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Building-frontend-angular-application.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+```sh
+npm install -g ng-openapi-gen
+npm install -g @angular/cli
+```
+
+You can check out [Angular documentation](https://angular.io/docs) for more
+details.
+
+2. Install [Angular CLI](npm install -g @angular/cli) by running
```
Comment (raymondfeng) [inline docs/site/Building-frontend-angular-application.md:30]: Should this line be removed?

File: docs/site/sidebars/lb4_sidebar.yml [hunk 2]
```diff
@@ -251,6 +251,13 @@ children:
         url: Using-openapi-to-graphql.html
         output: 'web, pdf'
 
+  - title: 'Building frontend application from LoopBack'
```
Comment (raymondfeng) [inline docs/site/sidebars/lb4_sidebar.yml:254]: Building frontend applications?

File: docs/site/Building-frontend-angular-application.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+keywords:
+  LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI, Angular, Frontend
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Building-frontend-angular-application.html
+---
+
+After a LoopBack application with REST APIs is created, the OpenAPI
+specification can be used to generate the models and web client in an Angular
+application using [ng-openapi-gen](https://www.npmjs.com/package/ng-openapi-gen)
+Node.js module.
```
Comment (raymondfeng) [inline docs/site/Building-frontend-angular-application.md:13]: It's probably better to set some context before jumping into `ng-openapi-gen`? Such as:

```
If you create REST APIs using LoopBack and need to build a Front-end application that consumes the APIs, ...
```

**Review by raymondfeng — APPROVED** [review]

File: docs/site/Building-frontend-angular-application.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+Use the `ng-openapi-gen` command to generate the models and web client for the
+Angular application based on the OpenAPI spec generated from the LoopBack
+application.
+
+```sh
+ng-openapi-gen --input <path-to-openapi-json> --output <angular-app-path>/src/app/api
+```
+
+You will see `src/app/api` folder is created, where the models are under the
+models folder and web client under the services folder.
```
Comment (jannyHou) [inline docs/site/Building-frontend-angular-application.md:62]: nitpick: 
```suggestion
`/models` folder and web client under the `/services` folder.
```

File: docs/site/Building-frontend-angular-application.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+command, and then a set of files in `src/app/<component-name>` will be created.
+Two files need to be modified:
+
+- `*/<component-name>.component.ts`: calls the REST APIs to retrieve data
+- `*/<component-name>.component.html`: displays the data
+
+Taking an example of the
+[todo application](https://loopback.io/doc/en/lb4/todo-tutorial.html),
+
+1. Updates `src/app/todolist/todolist.component.ts` get the data through the
```
Comment (jannyHou) [inline docs/site/Building-frontend-angular-application.md:83]:
```suggestion
1. Updates `src/app/todolist/todolist.component.ts` to get the data through the
```

**Review by jannyHou — APPROVED** [review]
👍 a few nitpicks.

**Pull request merged**